calico/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,48 @@
+name: calico
+files:
+  - name: calico.yaml
+    options:
+      - template: init_config
+        options:
+          - template: init_config/openmetrics
+      - template: instances
+        options:
+          - template: instances/openmetrics
+            required: true
+            description: Prometheus url and metrics to check
+            value:
+              type: string
+              example: http://localhost:9091/metrics
+          - name: metrics
+            required: true
+            description: "List of metrics to fetch"
+            value:
+              type: array
+              items:
+                type: string
+              example:
+                [
+                  "felix_active_local_endpoints",
+                  "felix_active_local_policies",
+                  "felix_active_local_selectors",
+                  "felix_active_local_tags",
+                  "felix_cluster_num_host_endpoints",
+                  "felix_cluster_num_hosts",
+                  "felix_cluster_num_workload_endpoints",
+                  "felix_ipset_calls",
+                  "felix_ipset_errors",
+                  "felix_ipsets_calico",
+                  "felix_ipsets_total",
+                  "felix_iptables_chains",
+                  "felix_iptables_rules",
+                  "felix_iptables_restore_calls",
+                  "felix_iptables_restore_errors",
+                  "felix_iptables_save_calls",
+                  "felix_iptables_save_errors",
+                ]
+          - name: namespace
+            required: true
+            description: "Namespace that will prefix every metric name."
+            value:
+              type: string
+              example: calico
```

Comment:
```suggestion
```

Reply:
I removed the metrics from the specs. But if I want my metrics to end up on datadog platform, I still have to indicate in the `conf.d/conf.yaml` a namespace, and the extra_metrics field containing all my felix metrics.
Should I actually try adding the namespace and the metrics in my check.py, so that the user does not configure it by himself ? In this case, I should modify each instance of the `instances` check parameter.

Reply:
I found a way to do it, using `get_default_config` to set metrics and also a namespace.